File: Python/01_Strings_Arrays/matrix_utils.py
import random

def compare(mat1, mat2, N, M):
    for i in range(0, N):
        for j in range(0, M):
            if mat1[i][j] != mat2[i][j]:
                return False
    return True


def print_matrix(mat, N, M):
    for i in range(0, N):
        for j in range(0, M):
            #print(str(i) + " " + str(j))
            print('{:02} '.format(mat[i][j]), end = '')
        print("\n")
    print("\n")
    return


def init_matrix(N, M):
    mat = [[i+N*j for i in range(M)] for j in range(N)]
    return mat

def init_matrix_with_list(N, M, li):
    count = 0
    mat = [[0 for i in range(M)] for j in range(N)]
    for i in range(0, N):
        for j in range(0, M):
            mat[i][j] = li[count]
            count+=1
    return mat

def init_rand_matrix(N, M):
    random.seed()

    mat = [[i+N*j for i in range(M)] for j in range(N)]#[[0]*M]*N
    for i in range(0, N):
        for j in range(0, M):
            
            value = random.randrange(N*M) % ((N*M)//2)
            #print("Value " + str(value) + " " + str(i) + " " + str(j))
            mat[i][j] = value
    #print_matrix(mat, N, M)
    return mat


def swap(mat, i1, j1, i2, j2):
    temp = mat[i1][j1]
    mat[i1][j1] = mat[i2][j2]
    mat[i2][j2] = temp
    return

File: Java/CTCI/src/strings_and_arrays/MatrixUtils.java
package strings_and_arrays;
import java.util.Random; 

public class MatrixUtils {
	
	public static int[][] init_matrix(int N, int M, int[] vals) {
		int[][] mat = new int[N][M];
		int count = 0;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < M; j++) {
				mat[i][j] = vals[count++];
			}
		}
		return mat;
	}
	
	public static int[][] init_zero_matrix(int N, int M) {
		int[][] mat = new int[N][M];

		for (int i = 0; i < N; i++) {
			for (int j = 0; j < M; j++) {
				mat[i][j] = 0;
			}
		}
		return mat;
	}
	
	public static int[][] init_matrix(int N, int M) {
		int[][] mat = new int[N][M];
		int count = 0;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < M; j++) {
				mat[i][j] = count++;
			}
		}
		return mat;
	}
	
	public static boolean compare(int[][] mat1, int[][] mat2, int N, int M) {
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < M; j++) {
				if (mat1[i][j] != mat2[i][j]) {
					return false;
				}
			}
		}
		return true;
	}
	public static int[][] init_matrix(int N) {
		return init_matrix(N, N);
	}
	
	public static int[][] init_rand_matrix(int N, int M) {
		int[][] mat = new int[N][M];
		Random rand = new Random(); 
		int randint;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < M; j++) {
				randint = rand.nextInt(1000);
				mat[i][j] = randint%(N*M/2);
			}
		}
		return mat;
	}
	
	public static void swap(int[][] mat, int i1, int j1, int i2, int j2) {
		int temp = mat[i1][j1];
		mat[i1][j1] = mat[i2][j2];
		mat[i2][j2] = temp;
	}
	
	public static void print_matrix(int[][] mat, int N, int M) {
		for (int i= 0; i < N; i++) {
			for (int j = 0; j < M; j++) {
				System.out.format("%2d ", mat[i][j]);
			}
			System.out.println();
		}
		System.out.println();
	}
	
	public static void print_matrix(int[][] mat, int N) {
		print_matrix(mat, N, N);
	}
}
